### DISCUSSION TITLE
Is support for python 3.7 needed?

### DISCUSSION THREAD

**Discussion by bl-young** [USEPA/flowsa#237] [Q&A] [answered]
Current implementation of the recursion and new FlowBy classes uses [Literal typing](https://peps.python.org/pep-0586/) which requires python 3.8.

Is there a need to maintain support for python 3.7? It will stop being supported in [June 2023](https://www.python.org/downloads/)

**Comment by catherinebirney** [comment]
For the sake of implementing all of Matthew's changes, I recommend dropping 3.7 when we pull in his branch.

**Reply by matthewlchambers** [reply]
While it would be easy enough to remove the use of Literal typing (since it's only used in the type hints and doesn't actually have any run-time effect), I also don't see a need to maintain support for Python 3.7.

**Comment by matthewlchambers** [comment, marked as answer]
Note that this relates to #238 . Given the discussion there, I propose we move to requiring `python >= 3.8`, `pandas >= 1.4`.

**Reply by matthewlchambers** [reply]
Marking this as answered (with a decision to require `python >= 3.8`, `pandas >= 1.4`) unless any objections are raised.

**Reply by catherinebirney** [reply]
I'll update the develop branch requirements

**Reply by bl-young** [reply]
because pandas is so widely used across the packages it might be good to do this on a separate branch so that it can be aligned with a specific release
